File: 백태균/week04/12940_최대공약수와_최소공배수.java
class Solution {
    public int[] solution(int n, int m) {

        int gcd = gcd(n, m);  // 유클리드 호제법으로 최대공약수 계산
        int lcm = (n * m) / gcd ;  // 최소공배수 = 두 수의 곱 / 최대공약수

        return new int[]{gcd, lcm};
    }

    // 유클리드 호제법으로 최대공약수(GCD) 구하기
    // a % b == 0이면 b가 최대공약수, 아니면 (b, a%b)로 범위를 줄여 재귀 호출
    static int gcd(int a, int b) {
        if (a % b == 0) {  // 나누어 떨어지면 나누는 수(b)가 최대공약수
            return b;
        }

        int r = a % b;  // 나머지 계산
        a = b;  // a를 b로 교체
        b = r;  // b를 나머지로 교체
        return gcd(a, b);   // 더 작은 수로 재귀 호출
    }
}

File: 김현우/week04/12940_최대공약수와_최소공배수.py
# 문제: 최대공약수와 최소공배수
# URL: https://school.programmers.co.kr/learn/courses/30/lessons/12940

def gcd(x, y):  # 유클리드 호제법
    while y:
        x, y = y, x%y
    return x

def lcm(x, y): return x*y/gcd(x, y)

def solution(n, m): return [gcd(n, m), lcm(n, m)]